sql_grader/xblocks.py

```diff
@@ -0,0 +1,119 @@
+"""
+A custom submission type to evaluate SQL queries
+"""
+import logging
+import os
+import pkg_resources
+
+from xblock.core import XBlock
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+from .models import SqlProblem
+from .mixins.dates import EnforceDueDates
+from .mixins.fragment import XBlockFragmentBuilderMixin
+from .mixins.grading import Scorable
+from .mixins.scenario import XBlockWorkbenchMixin
+from .models import XBlockDataMixin
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _(text):
+    """
+    Mock translation helper
+    """
+    return text
+
+
+def resource_string(path):
+    """
+    Handy helper for getting resources from our kit
+    """
+    data = pkg_resources.resource_string(__name__, path)
+    return data.decode('utf8')
+
+
+def all_datasets():
```

Comment:
Dynamically load all available datasets from disk (any `.sql` file in the `datasets` subdirectory.
This can be extended by operators at install/deploy time.